lib/rest-adapter.js

```diff
@@ -125,7 +125,7 @@ RestAdapter.prototype._extractAuth = function(remotes, invokeOptions) {
   return auth;
 };
 
-RestAdapter.prototype.invoke = function(method, ctorArgs, args, callback) {
+RestAdapter.prototype.invoke = function(method, ctorArgs, args, awOptions, callback) {
```

Comment:
I'm nervous about catching all the places that this might be called throughout the loopback framework, as it represents a breaking change.

An alternative approach might be to pull it off the object, e.g.

```js
RestAdapter.prototype.invoke = function( ...) {
  reqOptions = this.reqOptions || {};
  ...
```

It's a bit more hacky, but we could then have a utility for manually setting it via extracting the adapter.

If you think you can identify all places that this is used we can potentially keep it as is.

Reply:
The only parameter this could conflict with is the `callback` parameter which gets reassigned in the implementation to its correct value. So I think this change is non-breaking.

Reply:
Ah ok that was minimised when I looked so I missed it